target/ppc/translate/vsx-impl.c.inc

```diff
@@ -1546,6 +1546,42 @@ static bool trans_LXVKQ(DisasContext *ctx, arg_X_uim5 *a)
     return true;
 }
 
+static bool trans_XVTLSBB(DisasContext *ctx, arg_X_bf_xb *a)
+{
+    TCGv_i64 xb, all_true, all_false, crbf;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VSX(ctx);
+
+    xb = tcg_temp_new_i64();
+    all_true = tcg_const_i64(1);
+    all_false = tcg_const_i64(1);
+    crbf = tcg_temp_new_i64();
+
+    for (int dw = 0; dw < 2; dw++) {
+        get_cpu_vsr(xb, a->xb, dw);
+        for (int nbyte = 0; nbyte < 8; nbyte++) {
+            tcg_gen_and_i64(all_true, all_true, xb);
+            tcg_gen_andc_i64(all_false, all_false, xb);
+            tcg_gen_shri_i64(xb, xb, 8);
+        }
+    }
+
+    tcg_gen_mov_i64(crbf, all_true);
```

Comment:
By using the movcond mechanism above, you could also:
- initialize all_true with 0b1000 and all_false with 0b0010, instead of initializing them with 1
- then, you should be able to ``or`` all_true and all_false to get the contents of the CR field, without any additional shifting

Reply:
done, thanks